src/components/shared/thread/ActivityGraphFills.js

```diff
@@ -271,13 +282,29 @@ export class ActivityGraphFillComputer {
         : greyCategoryIndex;
 
     let cpuBeforeSample = null;
-    if (enableCPUUsage && threadCPUDelta && threadCPUDelta[lastIdx] !== null) {
-      const cpuDelta = threadCPUDelta[lastIdx];
-      const intervalDistribution =
-        lastIdx === 0
-          ? 1
-          : (samples.time[lastIdx] - samples.time[lastIdx - 1]) / interval;
-      cpuBeforeSample = cpuDelta / intervalDistribution;
+    let cpuAfterSample = null;
+    if (enableCPUUsage && threadCPUDelta) {
```

Comment:
I removed the `threadCPUDelta[lastIdx] !== null` check in this if and added an ensureExists instead. This should be okay here (we are doing the same in the previous loop too)